src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1060,6 +1085,49 @@ void SotaUptaneClient::verifySecondaries() {
   storage->storeMisconfiguredEcus(misconfigured_ecus);
 }
 
+bool SotaUptaneClient::waitSecondariesReachable(const std::vector<Uptane::Target> &updates) {
+  std::map<Uptane::EcuSerial, Uptane::SecondaryInterface *> targeted_secondaries;
+  const Uptane::EcuSerial &primary_ecu_serial = primaryEcuSerial();
+  for (const auto &t : updates) {
+    for (const auto &ecu : t.ecus()) {
+      if (ecu.first == primary_ecu_serial) {
+        continue;
+      }
+      auto f = secondaries.find(ecu.first);
+      if (f == secondaries.end()) {
+        LOG_ERROR << "Target " << t << " has unknown ECU ID";
+        continue;
+      }
+
+      targeted_secondaries[ecu.first] = f->second.get();
+    }
+  }
+
+  if (targeted_secondaries.empty()) {
+    return true;
+  }
+
+  LOG_INFO << "Waiting for secondaries to connect to start installation...";
+
+  // tries for 10 minutes
+  for (unsigned int tries = 1; tries < 600; tries++) {
+    if (targeted_secondaries.empty()) {
+      return true;
+    }
+
+    for (auto sec_it = targeted_secondaries.begin(); sec_it != targeted_secondaries.end();) {
+      if (sec_it->second->getManifest() != Json::Value()) {
```

Comment:
sec_it->second->getManifest() sets up a TCP connection with Secondary, a timeout of the setting up TCP connection really depends on multiple factors and configuration of TCP stack at a kernel level. The same is true for write/read to/from socket, so it won't be 10 minutes for sure and actually will significantly differs in a case of a local host and a non-local one (http://willbryant.net/overriding_the_default_linux_kernel_20_second_tcp_socket_connect_timeout)

Reply:
Yes that's true, I will rework something.

Reply:
BTW, why not just do certain amount of retries within getManifest() (I mean Asn1Rpc call) or keep retrying until certain time if we need to wait at least some period of time.